cpp/tests/streams/transform_test.cu

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/default_stream.hpp>
+#include <cudf_test/table_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/ast/expressions.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/detail/utilities/integer_utils.hpp>
+#include <cudf/transform.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/exec_policy.hpp>
+
+#include <cuda/functional>
+#include <thrust/fill.h>
+#include <thrust/functional.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/reduce.h>
+#include <thrust/tabulate.h>
+#include <thrust/transform.h>
+
+#include <numeric>
+
+class TransformTest : public cudf::test::BaseFixture {};
+
+std::pair<std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>>
+compute_segmented_row_bit_count(cudf::table_view const& input, cudf::size_type segment_length)
```

Comment:
Stream tests do not strictly need to check for output accuracy. The unit tests are meant to cover those. Stream tests are meant to simply check the stream parameter is passed correctly through to all kernel calls.
Therefore there is no real need for this elaborate function for checking the `segmented_row_bit_count` results.
And since the only reason for this file to be a .cu file is because this function uses thrust device code to verify, removing this function would allow it to become a .cpp file again.
My suggestion is just to call the `segmented_row_bit_count` in the `TEST_F(TransformTest, SegmentedRowBitCount)` below without checking the results. If the stream is used incorrectly, the test will automatically fail regardless of the results.
To ensure the compiler does not complain about unused results or optimizes out the function call you could wrap the call with a `EXPECT_NO_THROW()` perhaps.